File: LeetCode/C/88-two-pointers-合并两个有序数组.cpp
/*
 * @lc app=leetcode.cn id=88 lang=cpp
 *
 * [88] 合并两个有序数组
 */
#include <vector>
using namespace std;

// @lc code=start
class Solution {
  public:
    void merge(vector<int> &nums1, int m, vector<int> &nums2, int n) {
        int p = m + n - 1;
        int i = m - 1, j = n - 1;
        while (i >= 0 && j >= 0)
            nums1[p--] = nums1[i] > nums2[j] ? nums1[i--] : nums2[j--];
        while (j >= 0)
            nums1[p--] = nums2[j--];
    }
};
// @lc code=end

File: LeetCode/Go/88-two-pointers-合并两个有序数组.go
/*
 * @lc app=leetcode.cn id=88 lang=golang
 *
 * [88] 合并两个有序数组
 */
package leetcode

// @lc code=start
func merge(nums1 []int, m int, nums2 []int, n int) {
	var p = m + n - 1
	var i, j = m - 1, n - 1
	for i >= 0 && j >= 0 {
		if nums1[i] > nums2[j] {
			nums1[p] = nums1[i]
			i--
		} else {
			nums1[p] = nums2[j]
			j--
		}
		p--
	}
	for  j >= 0 {
		nums1[p] = nums2[j]
		p--
		j--
	}
}

// @lc code=end
